src/documentation/how-to-write-documentation.md

```diff
@@ -0,0 +1,149 @@
+# How to write documentation
+
+This document explains how to write documentation for the std/core public APIs.
+
+Let's start with some general information:
+
+### Contractions
+
+It is common in English to have contractions such as "don't" or "can't". Do not
+use these in documentation. Always write their "full form":
+
+ * "do not" instead of "don't"
+ * "cannot" instead of "can't"
+ * "it would" instead of "it'd"
+ * "it will" instead of "it'll"
+ * "it is"/"it has" instead of "it's"
+ * "you are" instead of "you're"
+ * "they are" instead of "they're"
+ * etc
+
+The only exception to this rule is "let's" as it is specific/known/common enough.
+
+The reason is simply to make the reading simpler for as many people as possible.
+
+### When to use inline code blocks
+
+Whenever you are talking about a type or anything code related, it should be in a
+inline code block. As a reminder, a inline code block is created with backticks
+(\`). For example:
+
+
+```text
+This a `Vec` and it has a method `push` which you can call by doing `Vec::push`.
+```
+
+### When to use intra-doc links
+
+Intra-doc links (you can see the full explanations for the feature
+[here](https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html))
+should be used as much as possible whenever a type is mentioned.
+
+Little note: when you are documenting an item, no need to link to it. So if you
+write documentation for `String::push_str` method, no need to link to the method
+`push_str` or to the `String` type.
+
+If you have cases like `Vec<String>`, you need to use intra-doc links on both
+`Vec` and `String` as well. It would look like this:
+
+```text
+This is a [`Vec`]`<`[`String`]`>`.
+```
```

Comment:
It makes the reading better. As such, I think it's a good enough reason.

(It's also something we enforced when the documentation team still existed too)

Reply:
> It makes the reading better.

It don't think it does. It just adds more noise with the changes in color/formatting.

If both `A` and `B` are very relevant, then it might make sense to link them separately in `A<B>`. But quite often `B` isn't very relevant, and it's fine and less noisy to just link all of `A<B>` to `A`.

Regardless, this is guideline is not uncontroversial. It's fine to document and merge guidelines in to std-dev-guide that we already follow, by documenting existing practice. But new rules that we don't currently follow should first have some discussion and a decision.

Right now, we have exactly _one_ doc comment that uses the syntax ``  [..]`<`[..]`>` ``.